File: solution/java/week01/exercise/src/main/java/movie/StoreAccount.java
package movie;

import java.util.ArrayList;
import java.util.List;

public class StoreAccount {
    public Double totalSold = 0d;

    //Implementation leak: MovieSale inner class exposed
    public List<MovieSale> allSales;

    public StoreAccount() {
        this.allSales = new ArrayList<>();
    }

    //Consufing naming: is 'to' the customer name?
    public void sell(Movie movie, String to) {
        totalSold += movie.unitPrice;
        //Bug: if adding to the list fail, the total is increased regardless.
        allSales.add(new MovieSale(to, movie.title));
    }

    static class MovieSale {
        String clientName;
        String movieName;

        public MovieSale(String clientName, String movieName) {
            this.clientName = clientName;
            this.movieName = movieName;
        }
    }
}

File: solution/c#/week01-code-analysis/ExerciseMovieStore/StoreAccount.cs
namespace ExerciseMovieStore;

public class StoreAccount
{
    public double TotalSold { get; set; } = 0d;
    //Implementation leak: MovieSale inner class exposed
    public List<MovieSale> AllSales { get; set; }

    public StoreAccount()
    {
        AllSales = new List<MovieSale>();
    }

    //Consufing naming: is 'to' the customer name?
    public void Sell(Movie movie, string to)
    {
        TotalSold += movie.UnitPrice;
        //Bug: if adding to the list fail, the total is increased regardless.
        AllSales.Add(new MovieSale(to, movie.Title));
    }

    public class MovieSale
    {
        public string ClientName { get; set; }
        public string MovieName { get; set; }

        public MovieSale(string clientName, string movieName)
        {
            ClientName = clientName;
            MovieName = movieName;
        }
    }
}